<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME)" />

  <arg name="standalone" default="true" />
  <arg name="debug" default="false" />
  <arg name="custom_config" default="" />
  <arg name="node_name" default="pcl_livox_to_ros" />
  <arg name="name_suffix" default="" />

  <arg name="topic_lidar_in"      default="livox/lidar" />
  <arg name="topic_lidar_out"     default="livox/lidar_ros" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <!-- Standalone -->
  <arg if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />

  <!-- Nodeleted -->
  <arg name="nodelet_manager_name" default="$(arg UAV_NAME)_pcl_nodelet_manager" />
  <arg if="$(eval not arg('standalone') and not arg('debug'))" name="nodelet" value="load" />
  <arg if="$(eval not arg('standalone') and not arg('debug'))" name="nodelet_manager" value="$(arg nodelet_manager_name)" />

  <arg if="$(eval arg('name_suffix') == '')" name="nodelet_name" value="$(arg node_name)" />
  <arg unless="$(eval arg('name_suffix') == '')" name="nodelet_name" value="$(arg node_name)_$(arg name_suffix)" />

  <group ns="$(arg UAV_NAME)">

    <node pkg="nodelet" type="nodelet" name="$(arg nodelet_name)" args="$(arg nodelet) mrs_pcl_tools/LivoxPCLtoPCL2 $(arg nodelet_manager)" output="screen" launch-prefix="$(arg launch_prefix)">

      <!-- Subscribers -->
      <remap from="~lidar_in" to="$(arg topic_lidar_in)" />

      <!-- Publishers -->
      <remap from="~lidar_out" to="$(arg topic_lidar_out)" />

    </node>

  </group>

</launch>
